mesonbuild/compilers/mixins/clike.py

```diff
@@ -110,16 +110,21 @@ def to_native(self, copy: bool = False) -> T.List[str]:
                 # Remove the -isystem and the path if the path is a default path
                 if (each == '-isystem' and
                         i < (len(new) - 1) and
-                        os.path.realpath(new[i + 1]) in real_default_dirs):
+                        self._cached_realpath(new[i + 1]) in real_default_dirs):
                     bad_idx_list += [i, i + 1]
-                elif each.startswith('-isystem=') and os.path.realpath(each[9:]) in real_default_dirs:
+                elif each.startswith('-isystem=') and self._cached_realpath(each[9:]) in real_default_dirs:
                     bad_idx_list += [i]
-                elif os.path.realpath(each[8:]) in real_default_dirs:
+                elif self._cached_realpath(each[8:]) in real_default_dirs:
                     bad_idx_list += [i]
             for i in reversed(bad_idx_list):
                 new.pop(i)
         return self.compiler.unix_args_to_native(new._container)
 
+    @staticmethod
+    @functools.lru_cache()
```

Comment:
Since these are used specifically for include paths, and there aren't that many, it probably makes sense to use `functools.cache` even.

Reply:
... which is in 3.9 only, so that would be `@functools.lru_cache(max_size=None)`.

Reply:
Done